w-12652328-adapting download tool 4.4 (#2477)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/_partials/upgrade-tool.adoc

//This content is reused in:
//mule-upgrade-tool.adoc

// Before You Begin
// tag::BeforeYouBegin[]

* Ensure that you have xref:release-notes::mule-upgrade-tool/mule-upgrade-tool.adoc[the latest available version of Mule upgrade tool] so that the tool runs with the latest fixes and security enhancements.
* A currently operational Mule 4 instance in _stopped_ status to prepare for the upgrade.
+

> For upgrades of Mule versions between 4.1.1 and 4.1.4, you must upgrade from any patch update released after January 20, 2022. Releases of versions 4.1.1 through 4.1.4, _before_ January 20, 2022 are not supported, and attempts to upgrade them produce an error message stating that the Mule version cannot be upgraded without first upgrading to a supported version (see xref:release-notes::mule-runtime/upgrade-update-mule.adoc[]).

Reviewer: ```suggestion
For upgrades of Mule versions between 4.1.1 and 4.1.4 with the Mule upgrade tool, you must upgrade from any patch update released after January 20, 2022. Releases of versions 4.1.1 through 4.1.4 _before_ January 20, 2022 are not supported by the tool, and attempts to upgrade them produce an error message stating that the Mule version cannot be upgraded without first upgrading to a supported version (see xref:release-notes::mule-runtime/upgrade-update-mule.adoc[]).
```

* The latest Mule runtime engine distribution, downloaded and unpacked in your system.
+

> The Mule upgrade tool requires the full distribution of the latest Mule runtime minor or patch version. Ensure you download a full Mule runtime distribution, identified by including the term `distribution` (and not the term `patch`) in the file name, for example, for the full distribution of the 20220221 patch for Mule 4.4.0: `mule-ee-distribution-standalone-4.4.0-20220221.zip`.

Reviewer: ```suggestion
The Mule upgrade tool requires the full distribution of the Mule runtime. Ensure you download a full Mule runtime distribution, identified by including the term `distribution` (and not the term `patch`) in the file name, for example, for the full distribution of the 20220221 patch for Mule 4.4.0: `mule-ee-distribution-standalone-4.4.0-20220221.zip`.
```

//This content is reused in:
//mule-upgrade-tool.adoc

// Before You Begin
// tag::BeforeYouBegin[]

* Ensure that you have xref:release-notes::mule-upgrade-tool/mule-upgrade-tool.adoc[the latest available version of Mule upgrade tool] so that the tool runs with the latest fixes and security enhancements.

> * A currently operational Mule instance running version 4.1.1 or later, in _stopped_ status to prepare for the upgrade.

Reviewer: Is Mule running when in a stopped status?

Author: ```suggestion
* A currently operational Mule 4 instance in _stopped_ status to prepare for the upgrade.
```

Reviewer: lgtm, @dmerlob

+

> When updating your current Mule instance from a version between 4.1.1 and 4.1.4, use any patch update released after January 20, 2022. If the running version is not one of the previously mentioned, an error message appears stating that the Mule version cannot be upgraded unless an upgrade is made.

Reviewer: Sorry, I still don't get this. 

Are they updating to a patch released after Jan 20, 2022?

Reviewer: And I don't know what this means: "If the running version is not one of the previously mentioned, an error message appears stating that the Mule version cannot be upgraded unless an upgrade is made."

> When updating your current Mule instance from a version between 4.1.1 and 4.1.4, use any patch update released after January 20, 2022. If the running version is not one of the previously mentioned, an error message appears stating that the Mule version cannot be upgraded unless an upgrade is made.

Reviewer: ```suggestion
For upgrades of Mule versions between 4.1.1 and 4.1.4, you must upgrade from any patch update released after January 20, 2022. Releases of versions 4.1.1 through 4.1.4, _before_ January 20, 2022 are not supported, and attempts to upgrade them produce an error message stating that the Mule version cannot be upgraded without first upgrading to a supported version (see xref:release-notes::mule-runtime/upgrade-update-mule.adoc[]).
```

Reviewer: Are we saying something like the above? Reading "use any patch update," I wondered how we were supposed to use it..

Reviewer: And if so, do we need to point people to other docs for upgrading those unsupported versions to a supported one?

Reviewer 2: actually versions between 4.1.1 and 4.1.4 are supported for versions released after january 2022, so if you have any of those versions, with those dates on it, you will be able to update. For prior versions you will need to follow the manual procedure that was part of the original document. Perhaps we need to add that link here.

Reviewer: Yes, can we add the link. Otherwise, the suggested text is saying exactly what you are saying. For those releases, you have to upgrade from a patch after after January 20, 2022, but attempting to upgraded from EARLIER releases within 4.1.1 through 4.1.4  throw an error. The previous description could be understood in different ways (one correct, another not), so was hoping to clarify. @jbianco

Reviewer: @dmerlob I added a link in the commit suggestion above.

> * The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory, that is, your old or existing Mule instance. 

Reviewer: Is it in the tool directory in a pre-v4.5 version of Mule? If not, should we remove it from the v4.4 version?

Reviewer 2: it will not be in versions pre-v4.5, son you should remove this part from v4.4 version

Author: Not directly. They have to download the tool by accessing the Help Center. Once the toll is downloaded, it can be found in the directory.

Author: Deleting section:
`* The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory, that is, your old or existing Mule instance.
+
To download the latest version of the tool (recommended), log in to https://help.mulesoft.com/s/[Help Center^]. Under the Support menu, select *Downloads > Mule Runtime*.
+
An example of the file name is `mule-runtime-upgrade-tool-1.1.0.zip`.`

> * The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory, that is, your old or existing Mule instance. 

Reviewer: ```suggestion
```

Reviewer: @dmerlob removing per Julio

+
To download the latest version of the tool (recommended), log in to https://help.mulesoft.com/s/[Help Center^]. Under the Support menu, select *Downloads > Mule Runtime*.
+
An example of the file name is `mule-runtime-upgrade-tool-1.1.0.zip`.

> * The latest Mule runtime engine distribution, downloaded and unpacked in your system.

Reviewer: So we are telling them to upgrade to the latest version of Mule and not to some earlier version of Mule? I'm not sure how to reconcile that with what's above:

"When updating your current Mule instance from a version between 4.1.1 and 4.1.4, use any patch update released after January 20, 2022."

> * The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory, that is, your old or existing Mule instance. 
> +
> To download the latest version of the tool (recommended), log in to https://help.mulesoft.com/s/[Help Center^]. Under the Support menu, select *Downloads > Mule Runtime*.
> +
> An example of the file name is `mule-runtime-upgrade-tool-1.1.0.zip`.

Author: ```suggestion
```

+

> The Mule upgrade tool requires the full distribution of the latest Mule runtime minor or patch version. Ensure you download a full Mule runtime distribution, identified by including the term `distribution` (and not the term `patch`) in the file name, for example, for the full distribution of the 20220221 patch for Mule 4.4.0: `mule-ee-distribution-standalone-4.4.0-20220221.zip`.

Reviewer: ok, so they always have to upgrade to the very latest version of Mule?

Reviewer 2: Not necessarily, you need to upgrade to a higher and/or newer version, but the tool is not checking the if the version is the latest.

Reviewer: Maybe then we should soften to recommend that people upgrade to the latest version but make clear that they can upgrade to any later version then?  @jbianco
